client/blocks/eligibility-warnings/test/index.tsx

```diff
@@ -170,7 +171,7 @@ describe( '<EligibilityWarnings>', () => {
 		);
 	} );
 
-	it( `disables the "Continue" button if holds can't be handled automatically`, () => {
+	it( `disables the "Continue" button if holds can't be handled automatically`, async () => {
```

Comment:
As the API `userEvent` provides is async (methods all return Promises) I think we should await them as the docs say.

In this case, the test passes because we test that `handleProceed` doesn't get called in case of a click. As we don't await the click, there may be no click happening at all which also makes the test pass. I'll add a fix and add the missing `await`.

Reply:
This is a really great catch by the way 👍🏻

Reply:
Adding an `await` makes sense 👍  Thanks for the confirmation.

Reply:
Does `userEvent.click` really return a promise? Looking at the `@testing-library/user-event` internals, I see all functions return `void`.

When performing a DOM click, there should be nothing async inside. Internally wrapped with `act()`, all React updates happen synchronously.

Reply:
I might be missing something, but I can clearly see the `click` convenience function returns a `Promise<void>`: 

https://github.com/testing-library/user-event/blob/e2a5f434b6e8b0ec0badddec52b54957b6f3320a/src/convenience/click.ts#L4

Reply:
Release notes for v14 mention that all APIs return Promises https://github.com/testing-library/user-event/releases/tag/v14.0.0 – relevant PR is https://github.com/testing-library/user-event/pull/790 (which closes https://github.com/testing-library/user-event/issues/504), those might contain more in-depth information about that change.